### PR TITLE
Allow deprecations to be configured & Split faraday-sunset

### PR DESCRIPTION
**Pull request by philsturgeon**

### CODE REVIEW

File: CHANGELOG [hunk 1]
```diff
@@ -7,6 +7,11 @@ and this project adheres to [Semantic Versioning](http://semver.org/).
 
 ## Unreleased
 
+### Changed
+- Switched `config.detect_deprecations` from bool to expect `:active_support` or logger instance
+
+## v0.5.0
```
Comment (bagmangood) [inline CHANGELOG]: do we want to up the version with this merge or are we releasing later?

File: lib/we/call/middleware/client/detect_deprecations.rb [hunk 2]
```diff
@@ -42,13 +42,21 @@ def report_deprecated_usage(env, datetime)
...
+              warned = true
+            end
+            if @options[:logger] && @options[:logger].respond_to?(:warn)
               @options[:logger].warn(warning)
-            else
-              raise NoOutputForWarning, "Pass active_support: true or logger: SomeLoggerWithWriteMethod when registering middleware"
+              warned = true
+            end
+            unless warned
+              raise NoOutputForWarning, "Pass active_support: true, or logger: ::Logger.new when registering middleware"
```
Comment (bagmangood) [inline lib/we/call/middleware/client/detect_deprecations.rb]: wouldn't this be an output we could detect at config load?

**Comment by philsturgeon** [comment]
Good eye, but this file is going to be a totally generic faraday middleware soon enough. I'm keeping it clean from we-call specific config. 

--
Phil Sturgeon

File: lib/we/call/middleware/client/detect_deprecations.rb (hunk 2 above)
Comment (philsturgeon) [reply lib/we/call/middleware/client/detect_deprecations.rb]: Followup commit has actually moved this file to its own faraday-sunset gem, so this won't be a thing anymore.

File: lib/we/call/middleware.rb [hunk 3]
```diff
@@ -1,7 +1,6 @@
 module We
   module Call
     module Middleware
-      autoload :Client, 'we/call/middleware/client'
```
Comment (philsturgeon) [inline lib/we/call/middleware.rb]: We don't have any client middlewares right now. If we make some, this'll come back.

File: README.md [hunk 4]
```diff
@@ -154,10 +154,12 @@ _For now this gem requires Rails 4.2+ due to some ActiveController functionality
 
 - [ ] Support adding href to Deprecate to make a `Link` with rel=sunset as per Sunset RFC draft 03
 - [ ] Remove Rails as a dependency (soft requirement on `ActiveSupport::Deprecated` is fine)
-- [ ] Split DetectDeprecations into standalone `faraday-sunset` gem
+- [x] Split DetectDeprecations into standalone [faraday-sunset] gem
 - [ ] Pass Trace IDs along
 - [ ] Work on sane defaults for retries and error raising
```
Comment (johnbrooks) [inline README.md]: Do we need to add Throttling to this.

**Pull request merged**